### PR TITLE
Update sights component to match design

### PR DESCRIPTION
**Pull request by thomasthesecond**
This pull request updates the sights component to match the design, but also has been refactored to remove complexity. The layout now uses flexbox.

One thing that I don't like is how `sights__item` is repeated 3 times in sights.hbs. I've created sights_item.hbs, but I can't include the partial inside of an `{{#each}}`. 

What I would like to do is this:

```handlebars
{{#each first_column}}
  {{> "components/sights/sights_item" }}
{{/each}}
```

But that causes the following error in Ruby:

```
undefined method `render' for #<OpenStruct …
```

Any suggestions?

### CODE REVIEW

**Comment by thomasthesecond** [comment]
I'll load the sights_item template client-side.

**Comment by thomasthesecond** [comment]
Related: https://github.com/lonelyplanet/destinations-next/pull/601

File: sass/elements/_buttons_mixins.scss [hunk 1]
```diff
@@ -30,8 +30,8 @@
 
   [class*="icon-"] {
     display: inline-block;
-    font-size: .6rem;
-    margin-left: .6rem;
+    font-size: 6px;
```
Comment (jcreamer898) [inline sass/elements/_buttons_mixins.scss]: What's up w/ no moar rem's here?

**Comment by jcreamer898** [comment]
Looks good. Just make sure we get some good testing done on no flexbox browsers.

File: sass/elements/_buttons_mixins.scss (hunk 1 above)
Comment (thomasthesecond) [inline sass/elements/_buttons_mixins.scss]: Mentioned in commit https://github.com/lonelyplanet/rizzo-next/commit/e10a73b6a8ee0939b0b660bda282b1c566edcc86.

REM: http://cl.ly/2M220N1m243s
Pixels: http://cl.ly/0G3Y1Z1B0E0M

**Pull request closed without merging**